### PR TITLE
fix: add support for node 12

### PR DESCRIPTION
**Pull request by mbroadst**
NODE-1984

### CODE REVIEW

File: src/kerberos_common.h [hunk 1]
```diff
@@ -26,19 +26,33 @@ inline void ResultDeleter(krb_result* result) {
 NAN_INLINE std::string StringOptionValue(v8::Local<v8::Object> options, const char* _key) {
     Nan::HandleScope scope;
     v8::Local<v8::String> key = Nan::New(_key).ToLocalChecked();
-    return !options.IsEmpty() && options->Has(key) && options->Get(key)->IsString()
-               ? std::string(*(Nan::Utf8String(options->Get(key))))
-               : std::string();
+    if (options.IsEmpty() || !Nan::Has(options, key).FromJust()) {
```
Comment (daprahamian) [inline src/kerberos_common.h]: Wouldn't it be safer to do use `FromMaybe(false)` here? The implication of the [`Nan::Maybe` docs](https://github.com/nodejs/nan/blob/master/doc/maybe_types.md#nanmaybe) is that `FromJust` has the potential to crash.

File: src/kerberos.cc [hunk 2]
```diff
@@ -131,7 +131,7 @@ NAN_GETTER(KerberosServer::ContextCompleteGetter) {
 
 NAN_METHOD(TestMethod) {
     std::string string(*Nan::Utf8String(info[0]));
-    bool shouldError = info[1]->BooleanValue();
+    bool shouldError = Nan::To<bool>(info[1]).FromJust();
```
Comment (daprahamian) [inline src/kerberos.cc:134]: Same question about `FromJust` vs. `FromMaybe` below.

File: src/kerberos_common.h (hunk 1 above)
Comment (mbroadst) [reply src/kerberos_common.h]: :+1:  In this case I think it's highly unlikely that we ever end up without a value and crash, but it seems fine to be extra careful

File: src/kerberos.cc (hunk 2 above)
Comment (mbroadst) [reply src/kerberos.cc:134]: :+1: This is definitely a case where it _could_ crash, but again we never let people directly call into this method, so unlikely. Changed it to be safer

**Pull request merged**